specs/CDPSessionIdSupport.md

```diff
@@ -0,0 +1,328 @@
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_targetInfoChangedToken));
+    // Enable Runtime events for the default target of top page to receive Runtime.consoleAPICalled events, which is fired when console.log is called.
+    m_webView->CallDevToolsProtocolMethod(L"Runtime.enable", L"{}", nullptr);
+    // Auto attach to iframe and dedicated worker targets created from the default target of top page.
+    m_webView->CallDevToolsProtocolMethod(
+        L"Target.setAutoAttach",
+        LR"({"autoAttach":true,"waitForDebuggerOnStart":false,"flatten":true})", nullptr);
+}
+```
+## WinRT and .NET    
+```c#
+// The sessions and targets descriptions are tracked by 2 dictionaries:
+// SessionId to TargetId dictionary: m_devToolsSessionMap;
+// TargetId to description dictionary, where description is "<target type>,<target url>": m_devToolsTargetInfoMap
+// GetJSONStringField is a helper function that can retrieve a string field from a json message.
+
+private void CoreWebView2_ConsoleAPICalled(CoreWebView2 sender, CoreWebView2DevToolsProtocolEventReceivedEventArgs args)
+{
+   // Figure out which target the console.log comes from
+   string eventSource;
+   string sessionId = args.SessionId;
+   if (sessionId.Length > 0)
+   {
+     string targetId = m_devToolsSessionMap[sessionId];
+     eventSource = m_devToolsTargetDescriptionMap[targetId];
+   }
+  // App code to log these events.
+  LogConsoleLogMessage(eventSource, args.ParameterObjectAsJson);
+}
+
+private void CoreWebView2_AttachedToTarget(CoreWebView2 sender, CoreWebView2DevToolsProtocolEventReceivedEventArgs args)
+{
+  // A new target is attached, add its info to maps.
+  string jsonMessage = args.ParameterObjectAsJson;
+  string sessionId = GetJSONStringField(jsonMessage, L"sessionId");
+  string targetId = GetJSONStringField(jsonMessage, L"targetId");
+  m_devToolsSessionMap[sessionId] = targetId;
+  string type = GetJSONStringField(jsonMessage, L"type");
+  string url = GetJSONStringField(jsonMessage, L"url");
+  m_devToolsTargetDescriptionMap[targetId] = type + L"," + url;
+  // Auto attach to targets further created from this target.
+  _ = m_webview.CallDevToolsProtocolMethodAsync("Target.setAutoAttach",
+        @"{""autoAttach"":true,""waitForDebuggerOnStart"":false,""flatten"":true}",
+        sessionId);
+  // Enable Runtime events to receive Runtime.consoleAPICalled from the target, which is triggered by console.log calls.
+  m_webview.CallDevToolsProtocolMethodAsync("Runtime.enable", "{}", sessionId);
+}
+
+private void CoreWebView2_DetachedFromTarget(CoreWebView2 sender, CoreWebView2DevToolsProtocolEventReceivedEventArgs args)
+{
+  // A target is detached, remove it from the maps
+  string jsonMessage = args.ParameterObjectAsJson;
+  string sessionId = GetJSONStringField(jsonMessage, L"sessionId");
+  if (m_devToolsSessionMap.ContainsKey(sessionId))
```

Comment:
Simpler is

```csharp
if (m_devToolsSessionMap.Remove(sessionId, out string targetId))
{
    m_devToolsTargetDescriptionMap.Remove(targetId);
}
```

Probes `m_devToolsSessionMap` only once instead of three times.

Reply:
Please update.